content/data-loading.md

```diff
@@ -0,0 +1,556 @@
+2. The subscription detects this, and given that anything is possible in next computation run, marks itself for destruction.
+3. The computation re-runs, with `.subscribe()` being re-called either with new or different arguments.
+4a. If the subscription is run with the *same arguments* then the "new" subscription discovers the old "marked for destruction" subscription that sitting around, with the same data already ready, and simply reuses that.
+4b. If the subscription is run with *different arguments*, then a new subscription is created, which connects to the publication on the server
+5. At the end of the flush cycle (i.e. after the computation is done re-running), the old subscription checks to see if it was re-used, and if not, sends a message to the server to tell the server to shut it down.
+
+The important detail in the above is in 4a --- that they system cleverly knows not to re-subscribe if the autorun re-runs and subscribes with the exact same arguments. This holds true even if the new subscription is set up somewhere else in the template heirarchy.
+
+For instance if a user navigates between two pages that both subscribe to the exact same subscription, the same mechanism will kick in and no unnecessarily subscribing will happen.
+
+### Publication behaviour when changing arguments
+
+It's also worth knowing a little about what happens on the server when the new subscription is started and the old one is stopped.
+
+The server *explicitly* waits until all the data is sent down (the new subscription is ready) for the new subscription before removing the data from the old subscription. The idea here is to avoid flicker -- you can, if desired, continue to show the old subscription's data until the new data is ready, then instantly switch over to the new subscription's complete data set.
+
+What this means is in general, when changing subscriptions, there'll be a period where you are *over-subscribed* and there is more data on the client than you strictly asked for. This is one very important reason why you should always fetch the same data that you have subscribed to (don't "over-fetch").
+
+### Paginating subscriptions
+
+A very common pattern of data access is pagination. This refers to the practice of fetching a ordered list of data one "page" at a time --- typically some number of items, say twenty.
+
+There are two styles of pagination that are commonly used, a "page-by-page" style---where you show only one page of results at a time, starting at some offset (which the user can control), and "infinite-scroll" style, where you show and increasing number of pages of items, as the user moves through the list (this is the typical "feed" style user interface).
+
+Let's consider a publication/subscription technique for the second (the first technique is quite difficult to do correctly in Meteor as you can't tell at which offset to start on the client-side which only has a subset of the true data).
```

Comment:
Yikes! Is there at least somewhere we can link to that tells you how to do it correctly? Or is it literally impossible?